### PR TITLE
Followup to #24844

### PR DESCRIPTION
**Pull request by maclover7**
Some slight documentation edits and fixes. Also, run remove unnecessary
`RuntimeError`.

r? @arthurnn

### CODE REVIEW

**Comment by kamipo** [comment]
#24844 ?

**Comment by maclover7** [comment]
whoops, yeah. thanks @kamipo! let me update the commit message, and then I think this is ready to go (test are passing 👍

File: activerecord/lib/active_record/connection_handling.rb [hunk 1]
```diff
@@ -93,8 +93,7 @@ def connection
 
     attr_writer :connection_specification_name
 
-    # Return the specification id from this class otherwise look it up
-    # in the parent.
+    # Return the specification id from the current class or its parent.
```
Comment (arthurnn) [inline activerecord/lib/active_record/connection_handling.rb]: s/specification id/specification name

File: activerecord/lib/active_record/connection_handling.rb [hunk 2]
```diff
@@ -45,7 +45,7 @@ module ConnectionHandling
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
     def establish_connection(spec = nil)
-      raise RuntimeError, "Anonymous class is not allowed." unless name
```
Comment (vipulnsward) [inline activerecord/lib/active_record/connection_handling.rb]: We can get rid of it from other places as well.

Comment (maclover7) [inline activerecord/lib/active_record/connection_handling.rb]: Looks like this is the only place in #24844

File: activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb [hunk 3]
```diff
@@ -909,10 +909,11 @@ def remove_connection(spec_name)
       # to optimise for.
       def retrieve_connection_pool(spec_name)
         owner_to_pool.fetch(spec_name) do
+          # Check if a connection was previously established in an ancestor process,
+          # which may have been worked.
```
Comment (jeremy) [inline activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb]: may have been *forked?

File: activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb [hunk 4]
```diff
@@ -909,10 +909,11 @@ def remove_connection(spec_name)
...
           if ancestor_pool = pool_from_any_process_for(spec_name)
-            # A connection was established in an ancestor process that must have
-            # subsequently forked. We can't reuse the connection, but we can copy
-            # the specification and establish a new connection with it.
+            # We can't reuse the connection, but we can copy the specification,
+            # and then establish a new connection.
```
Comment (jeremy) [inline activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb]: The original comment is clearer. Let's stick with it.

File: activerecord/lib/active_record/connection_handling.rb [hunk 5]
```diff
@@ -45,7 +45,7 @@ module ConnectionHandling
...
+      raise "Anonymous class is not allowed." unless name
```
Comment (jeremy) [inline activerecord/lib/active_record/connection_handling.rb]: 👍

File: activerecord/lib/active_record/connection_handling.rb [hunk 6]
```diff
@@ -93,8 +93,7 @@ def connection
 
     attr_writer :connection_specification_name
 
-    # Return the specification id from this class otherwise look it up
-    # in the parent.
+    # Return the specification name from the current class or its parent.
```
Comment (jeremy) [inline activerecord/lib/active_record/connection_handling.rb]: 👍

File: activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb (hunk 3 above)
Comment (maclover7) [inline activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb]: whoops, typo :)

**Comment by maclover7** [comment]
updated per your comments, @jeremy. thanks for the review! :)

File: activerecord/lib/active_record/connection_handling.rb (hunk 2 above)
Comment (vipulnsward) [inline activerecord/lib/active_record/connection_handling.rb]: It exists at 4-5 more places, AM, Fixtures, etc.

Comment (maclover7) [inline activerecord/lib/active_record/connection_handling.rb]: Happy to open up another PR with that -- wanted to try and keep this PR centered on a few fixes to Arthur's PR. :-)

I'll open up another PR and ping you on it for review.

Comment (arthurnn) [inline activerecord/lib/active_record/connection_handling.rb]: make sense.. thanks .

**Comment by arthurnn** [comment]
thanks

**Comment by maclover7** [comment]
👍 thanks @arthurnn!

**Pull request merged**